app/models/user.rb

```diff
@@ -0,0 +1,25 @@
+class User < ApplicationRecord
+  has_secure_password
+
+  has_many :tasks, dependent: :destroy
+
+  validates :name, presence: true, length: { maximum: 20 }
+  validates :email, presence: true, length: { maximum: 50 }, uniqueness: true,
+                    format: { with: /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i }
+  validates :password, presence: true, length: { minimum: 6 }, confirmation: true
+  validates :password_confirmation, presence: true
+
+  before_validation { email.downcase! }
+  before_update :not_update_last_admin
+  before_destroy :not_destroy_last_admin
+
+  private
+
+  def not_update_last_admin
+    throw(:abort) if User.where(admin: true).count == 1 && self.admin == false && self.id == User.find_by(admin: true).id
```

Comment:
この部分は両立しないと思われます。
`self.admin == false && self.id == User.find_by(admin: true).id`

下記の様に短く記載できると思うのでご確認ください。
```suggestion
    throw(:abort) if User.admin.one? && self.admin == false && self.id == User.find_by(admin: true).id
```

Reply:
1177ea2 で一部修正しました。

このコードでは、管理者権限を持つユーザーが1人だけの場合に、自分の管理者権限を編集できないように設定しています。

* `self.admin == false`は、ユーザー編集画面で管理者権限を削除する編集を行った場合
* `self.id == User.find_by(admin: true).id`は、編集対象が管理者権限を持つユーザーである場合

を指定しているので両立すると思うのですが、僕の認識で間違っているところがあるのでしょうか？

Reply:
こちら自分の認識が誤っていました。下記にコメントしております。
https://github.com/shohei-hotta/task_app/pull/4#discussion_r419880438